src/Disruptor/SingleProducerSequencer.cs

```diff
@@ -125,14 +124,15 @@ public long Next()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public long Next(int n)
         {
-            if (n < 1 || n > _bufferSize)
+            if (unchecked((uint)(n - 1)) >= _bufferSize)
```

Comment:
Why is unchecked used here?

Reply:
I have a habit of enabling int overflow checks in debug mode and being explicit when overflow is expected. Could be removed here.